app/models/metric/ci_mixin/state_finders.rb

```diff
@@ -1,16 +1,18 @@
 module Metric::CiMixin::StateFinders
+  # @param ts [Time|String] prefer Time
   def vim_performance_state_for_ts(ts)
+    ts = Time.parse(ts).utc if ts.kind_of?(String)
+    isots = ts.utc.iso8601
     return nil unless self.respond_to?(:vim_performance_states)
 
     @states_by_ts ||= {}
-    state = @states_by_ts[ts]
+    state = @states_by_ts[isots]
```

Comment:
So, just for my education, we were using Time or String objects as keys in this hash and now we're using Time exclusively.  Is that right?

Reply:
### before:

```ruby
@states_by_ts[time.to_s] => VimPerformanceState
# sometimes that is iso8601, other times a different format
```

### After

```ruby
@states_by_ts[time.to_s] => VimPerformanceState
# always iso8601
```